torch/testing/__init__.py

```diff
@@ -199,6 +216,12 @@ def _helper(a, b, s) -> _compare_return_type:
         a = complex(a)
         b = complex(b)
 
+        if equal_nan == "relaxed":
```

Comment:
This seems like it will create a confusing error message in cases like 5 + 0j being compared with 5 + NaNj, because the latter number will become NaN + NaNj and the real part will compare as different, not the imaginary part. 

Instead, when equal_nan is relaxed just short-circuit future checks like this:

```
if a != a and b != b:
  return (True, None)
```